components/TextFields/src/Private/MDCTextInputCommonFundament.m

```diff
@@ -0,0 +1,944 @@
+
+  return underlineYConstant;
+}
+
+- (BOOL)needsUpdateUnderlinePosition {
+  return self.underlineY.constant != [self underlineYConstant];
+}
+
+- (void)updateUnderlinePosition {
+  self.underlineY.constant = [self underlineYConstant];
+  [self.textInput invalidateIntrinsicContentSize];
+}
+
+#pragma mark - Properties Implementation
+
+- (NSAttributedString *)attributedPlaceholder {
+  id placeholderString = self.placeholderLabel.text;
+  if ([placeholderString isKindOfClass:[NSString class]]) {
+    // TODO: (larche) Return string attributes also. Tho I feel like that should come from the
+    // placeholderLabel itself somehow.
+    NSAttributedString *constructedString =
+        [[NSAttributedString alloc] initWithString:(NSString *)placeholderString attributes:nil];
+    return constructedString;
+  } else if ([placeholderString isKindOfClass:[NSAttributedString class]]) {
+    return (NSAttributedString *)placeholderString;
+  }
+
+  return nil;
+}
+
+- (void)setAttributedPlaceholder:(NSAttributedString *)attributedPlaceholder {
+  self.placeholderLabel.text = attributedPlaceholder.string;
+  // TODO: (larche) Read string attributes also. Tho I feel like that should come from the
+  // placeholderLabel itself somehow.
+
+  [self updatePlaceholderAlpha];
+  [self.textInput setNeedsUpdateConstraints];
+}
+
+- (void)setClearButtonColor:(UIColor *)clearButtonColor {
+  if (![_clearButtonColor isEqual:clearButtonColor]) {
+    _clearButtonColor = clearButtonColor;
+    self.clearButtonImage =
+        [self drawnClearButtonImage:self.clearButtonImage.size color:_clearButtonColor];
+  }
+}
+
+- (void)setClearButtonMode:(UITextFieldViewMode)clearButtonMode {
+  _clearButtonMode = clearButtonMode;
+  [self updateClearButton];
+}
+
+- (void)setEnabled:(BOOL)enabled {
+  _enabled = enabled;
+  self.underline.enabled = enabled;
+}
+
+- (UIFont *)font {
+  return self.textInput.font;
+}
+
+- (void)setFont:(UIFont *)font {
+  [self.textInput setFont:font];
+}
+
+- (void)setHidesPlaceholderOnInput:(BOOL)hidesPlaceholderOnInput {
+  _hidesPlaceholderOnInput = hidesPlaceholderOnInput;
+  [self updatePlaceholderAlpha];
+}
+
+- (BOOL)isEditing {
+  return self.textInput.isEditing;
+}
+
+- (NSString *)placeholder {
+  id placeholderString = self.placeholderLabel.text;
+  if ([placeholderString isKindOfClass:[NSString class]]) {
+    return (NSString *)placeholderString;
+  } else if ([placeholderString isKindOfClass:[NSAttributedString class]]) {
+    return [(NSAttributedString *)placeholderString string];
+  }
+
+  return nil;
+}
+
+- (void)setPlaceholder:(NSString *)placeholder {
+  self.placeholderLabel.text = placeholder;
+  [self updatePlaceholderAlpha];
+  [self.textInput setNeedsLayout];
+}
+
+- (NSString *)text {
+  return self.textInput.text;
+}
+
+- (void)setText:(NSString *)text {
+  [self.textInput setText:text];
+}
+
+- (void)setTextColor:(UIColor *)textColor {
+  if (!textColor) {
+    textColor = MDCTextInputTextColor();
```

Comment:
Consider setting the default color values on the UIAppearance proxy in the +initialize method.